league/dashboard/models.py

```diff
@@ -3,35 +3,97 @@
 
 from enum import Enum
 
-from league.database import Column, Model, SurrogatePK, db, reference_col
+from league.database import (Column, Model, SurrogatePK, association_proxy, db,
+                             reference_col, relationship)
 
 Color = Enum('Color', 'white black')
 
 
+class Player(SurrogatePK, Model):
```

Comment:
Interesting approach, I hadn't thought about implementing it this way.

Having two distinct join tables seems strange to me, though. In theory, because the information is spread out over several tables, there could be inconsistencies. The way you did it before seems clearer and simpler. Why not just keep the old way, and add some helper functions to Player to execute the correct queries to stitch together their games played as white/black/both?

As an example of the inconsistencies: Let's say you have `Player1` and `Player2` and you create a game `Game17`. You do `Player1.white_player_games.append(Game17)`. If you forget to do `Player2.black_player_games.append(Game17)`, does it ever get recorded? Is it possible to have the same game played by multiple people? (i.e. a WhitePlayerGame join relationship between Player1/Game17, Player2/Game17, Player3/Game17...).  It seems like a more robust design to just say `Game17(white=Player1, black=Player2)`, and have everything just automatically work.